storehaus-postgresql/src/main/scala/com/twitter/storehaus/postgres/PostgresStore.scala

```diff
@@ -0,0 +1,178 @@
+package com.twitter.storehaus.postgres
+
+import com.twitter.finagle.postgres.Client
+import com.twitter.finagle.postgres.values.Value
+import com.twitter.storehaus.Store
+import com.twitter.util.{Future, Time}
+
+import scala.util.{Failure, Success}
+
+/**
+  * Key value storage with PostgreSQL (>9.5) as backend
+  *
+  * @author Alexey Ponkin
+  */
+object PostgresStore {
+
+  def apply[K, V]
+  (client: Client, table: String, kCol: String, vCol: String)
+  (implicit kInj: PostgresValueConverter[K], vInj: PostgresValueConverter[V]) =
+    new PostgresStore[K, V](client, table, kCol, vCol)
+}
+
+class PostgresStore[K, V]
+(protected[postgres] val client: Client, table: String, kCol: String, vCol: String)
+(implicit kInj: PostgresValueConverter[K], vInj: PostgresValueConverter[V])
+  extends Store[K, V] {
+
+  private val toV: PostgresValue => V = vInj.invert(_) match {
+    case Success(v) => v
+    case Failure(e) => throw IllegalConversionException(e.getMessage)
```

Comment:
why do we need to throw here? Why not just put this in the future?
```scala
private def toV(p: PostgresValue): Future[V] = vInj.invert(p) match {
  case Success(v) => Future.value(v)
  case Failure(e) => Future.exception(e)
}
```
then use `toV(v).flatMap { ...` rather than throwing?